src/test/groovy/com/pretz/parkingmanager/calculator/DuesCalculatorSpec.groovy

```diff
@@ -0,0 +1,230 @@
+package com.pretz.parkingmanager.calculator
+
+import com.pretz.parkingmanager.UnitTest
+import com.pretz.parkingmanager.calculator.currency.CurrencyConverter
+import com.pretz.parkingmanager.calculator.DuesCalculator
+import com.pretz.parkingmanager.calculator.hours.DefaultParkingHoursCalculator
+import com.pretz.parkingmanager.domain.ParkingRate
+import com.pretz.parkingmanager.domain.ParkingSession
+import org.junit.experimental.categories.Category
+import spock.lang.Specification
+
+import java.sql.Timestamp
+import java.time.Instant
+import java.time.temporal.ChronoUnit
+
+@Category(UnitTest.class)
+class DuesCalculatorSpec extends Specification {
+
+    private DuesCalculator duesCalculator
+    private CurrencyConverter currencyConverter
+
+    private ParkingSession.ParkingSessionBuilder testBuilder
+
+    def setup() {
+        currencyConverter = Stub(CurrencyConverter)
+        currencyConverter.getCurrencyMultiplier() >> BigDecimal.ONE
+
+        duesCalculator = new DuesCalculator(new DefaultParkingHoursCalculator())
+
+        testBuilder = ParkingSession.builder()
+    }
+
+    def "should calculate 1 PLN due for 'regular' parking stop shorter than 1 hour"() {
+
+        given:
+        ParkingSession testParkingSession = testBuilder
+                .startTime(Timestamp.from(Instant.now()
+                .minus(45, ChronoUnit.MINUTES)
+                .minus(11, ChronoUnit.SECONDS)))
+                .parkingRate(ParkingRate.REGULAR)
+                .build()
+
+        when:
+        BigDecimal testDues = duesCalculator.calculateDues(testParkingSession, currencyConverter)
+
+        then:
+        testDues == BigDecimal.valueOf(1.00)
```

Comment:
blok expect nie byłby lepszy?

Reply:
Może i byłby, ale lubię widoczne wydzielenie bloków given-when-then. Dlatego np. nie spodobało mi się, że nie można w Spocku rodzielić mockowania od wyrażenia opisującego liczbę oczekiwanych wywołań mockowanej metody (dostałem wyjaśnienie na stacku, ale nadal mi się to nie podoba):

https://stackoverflow.com/questions/53154147/spock-mocked-repository-method-save-giving-nullpointerexception